pkg/manager/impl/validation/shared_execution.go

```diff
@@ -0,0 +1,34 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flyteorg/flyteadmin/pkg/common"
+	"github.com/flyteorg/flyteadmin/pkg/errors"
+	"github.com/flyteorg/flyteadmin/pkg/repositories"
+	repoInterfaces "github.com/flyteorg/flyteadmin/pkg/repositories/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytestdlib/logger"
+)
+
+func ValidateCluster(ctx context.Context, db repositories.RepositoryInterface, executionID *core.WorkflowExecutionIdentifier, cluster string) error {
+	if len(cluster) == 0 || cluster == common.DefaultProducerID {
+		return nil
+	}
+	workflowExecution, err := db.ExecutionRepo().Get(ctx, repoInterfaces.Identifier{
+		Project: executionID.Project,
+		Domain:  executionID.Domain,
+		Name:    executionID.Name,
+	})
```

Comment:
We do already check if execution exists here https://github.com/flyteorg/flyteadmin/blob/296655a415d36e20bdedf98e41f890eb4901480c/pkg/manager/impl/node_execution_manager.go#L92 before calling ValidateCluster.

Can ValidateCluster be made independent of the DB call inorder to reduce the same checks on the DB or probably have a common refactored Validate function which does these checks

Reply:
sure, done